mlir/lib/Dialect/Rock/utility/loweringUtils.cpp

```diff
@@ -616,3 +618,62 @@ mlir::rock::transposeSubTileViews(PatternRewriter &rewriter, Location loc,
                                 std::nullopt};
   }
 }
+
+template <typename RetAttrType>
+static FailureOr<RetAttrType>
+getAttrFromOpOrParents(Operation *op, StringRef opAttr,
+                       std::optional<StringRef> maybeFuncAttr = std::nullopt,
+                       std::optional<StringRef> maybeModAttr = std::nullopt) {
```

Comment:
Since there are only 2 kinds of attr names, 
1) property/required attributes (not dialect qualified), and 
2) discardable attributes (which should always be dialect-qualified)

this can be simplified to have just one `maybeDialectAttr` for the discardable attr (on Func and Module).

Reply:
done!